### PR TITLE
Move Tensorboard Integration

### PR DESCRIPTION
**Pull request by dheemantha-bhat**
### Motivation
Progress optuna/optuna#4484  

**Description of the changes** 
Move integration/tensorboard.py and its test from optuna/optuna

### CODE REVIEW

File: docs/source/reference/index.rst
Comment (nzw0301) [inline docs/source/reference/index.rst:97]:
```suggestion
-----------
```

**Review by not522 — APPROVED** [review]
Thank you for your PR. LGTM!
Please wait until the second review for the merge.

**Comment by not522** [comment]
I'll merge this PR after https://github.com/optuna/optuna/pull/5161 is merged.

**Pull request merged**
